### PR TITLE
Refactor NPCRetaliationSystem::Update to fix crash caused by Debug Assertion

### LINKED ISSUE #21702: retaliation debug assert
## Description
using wrong uid for meta resolve
happens when mob retaliating dies

**Reproduction**
1. spawn goat
2. spawn knife
3. shank goat to death
4. boop

**Stacktrace**
```
Unhandled exception. Robust.Shared.Utility.DebugAssertException: Entity 18090 is not the owner of the component. Component: MetaDataComponent
   at Robust.Shared.Utility.DebugTools.AssertOwner(Nullable`1 uid, IComponent component) in RobustToolbox/Robust.Shared/Utility/DebugTools.cs:line 103
   at Content.Server.NPC.Systems.NPCRetaliationSystem.Update(Single frameTime) in Content.Server/NPC/Systems/NPCRetaliationSystem.cs:line 78
   at Robust.Shared.GameObjects.EntitySystemManager.TickUpdate(Single frameTime, Boolean noPredictions) in RobustToolbox/Robust.Shared/GameObjects/EntitySystemManager.cs:line 306
   at Robust.Shared.GameObjects.EntityManager.TickUpdate(Single frameTime, Boolean noPredictions, Histogram histogram) in RobustToolbox/Robust.Shared/GameObjects/EntityManager.cs:line 236
   at Robust.Server.GameObjects.ServerEntityManager.TickUpdate(Single frameTime, Boolean noPredictions, Histogram histogram) in RobustToolbox/Robust.Server/GameObjects/ServerEntityManager.cs:line 160
   at Robust.Server.BaseServer.Update(FrameEventArgs frameEventArgs) in RobustToolbox/Robust.Server/BaseServer.cs:line 722
   at Robust.Server.BaseServer.<SetupMainLoop>b__66_1(Object sender, FrameEventArgs args) in RobustToolbox/Robust.Server/BaseServer.cs:line 539
   at Robust.Shared.Timing.GameLoop.Run() in RobustToolbox/Robust.Shared/Timing/GameLoop.cs:line 235
   at Robust.Server.BaseServer.MainLoop() in RobustToolbox/Robust.Server/BaseServer.cs:line 566
   at Robust.Server.Program.ParsedMain(CommandLineArgs args, Boolean contentStart, ServerOptions options) in RobustToolbox/Robust.Server/Program.cs:line 78
   at Robust.Server.Program.Start(String[] args, ServerOptions options, Boolean contentStart) in RobustToolbox/Robust.Server/Program.cs:line 46
   at Robust.Server.ContentStart.Start(String[] args) in RobustToolbox/Robust.Server/ContentStart.cs:line 10
   at Content.Server.Program.Main(String[] args) in Content.Server/Program.cs
```

### PR DESCRIPTION
**Pull request by Chronophylos**
## About the PR
I refactored the `Update` function in the `NPCRetaliationSystem` to prevent a crash caused by Debug Assertion.
fixes #21702 

## Technical details

I eliminate the queue by combining the two foreach loops and remove the check for entity termination, as it only results in deaggroing the entity twice without altering anything else.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

### CODE REVIEW

**Comment by lzk228** [comment]
they'll kill you for webedit

**Comment by Chronophylos** [comment]
what do you mean?

**Comment by Partmedia** [comment]
> what do you mean?

Web edits (edits made through the Github web editor) are [not allowed by policy](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html#testing).

You signed your commits using your GPG key, which shows up on Github as "Verified", which people assume come from web editing. So you are free to ignore them :)

File: Content.Server/NPC/Systems/NPCRetaliationSystem.cs [hunk 1]
```diff
@@ -65,25 +63,15 @@ public override void Update(float frameTime)
...
-
-            foreach (var ent in new ValueList<EntityUid>(comp.AttackMemories.Keys))
+            foreach (var entity in new ValueList<EntityUid>(retaliationComponent.AttackMemories.Keys))
             {
-                if (_timing.CurTime < comp.AttackMemories[ent])
+                if (_timing.CurTime < retaliationComponent.AttackMemories[entity])
                     continue;
 
-                if (TerminatingOrDeleted(ent, metaData))
-                    _deAggroQueue.Add(ent);
```
Comment (EmoGarbage404) [inline Content.Server/NPC/Systems/NPCRetaliationSystem.cs:79]: You need to still remove terminating / deleted entities or else they'll throw errors when we try and access data on them. Just remove passing in the MetaData component and keep everything else the same.

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

**Comment by Chronophylos** [comment]
@Partmedia Thank you for elaborating :). Verifying my commits has been a habit for years.

**Review by EmoGarbage404 — APPROVED** [review]

**Comment by EmoGarbage404** [comment]
I opted to remove the changelog in accordance with the significance guidelines.

**Pull request merged**